[web] redirect HTTP -> HTTPS

adborden opened the pull request:
https://github.com/GSA/datagov-deploy/issues/1858

Redirect HTTP -> HTTPS for all load balancers. Should be safe for all apps
since this is how we're configurd in staging/production.

adborden committed:
[web] redirect HTTP -> HTTPS

jbrown-xentity reviewed (approved):
Makes sense and looks good, but not sure why build is failing due to this change... Leaving to @adborden  to debug and merge.

adborden committed:
Specify provider requirements

Instead of declaring a provider, specify the provider requirements. Works around
issue with aws provider v3.x and the alb resource[1].

[1]: https://github.com/GSA/datagov-deploy/issues/2032
